projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -1302,8 +1304,9 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
      */
     private generateDateRange(start: Date, end: Date): Date[] {
```

Comment:
In case when we have disabled dates and select them from the API if generateDateRange returns DateRangeDescriptor we will have to remove the disabled dates from this range and it will be a Specific DateRangeDescriptor  similar as an array as it is now.

Reply:
But why? Disabled dates can be selected programmatically, just cannot be interacted with.